commit b43bf038050c983cf7587395fd6fb24db0f275c3 (refs/changes/41/2841/4)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2017-10-25T09:22:00-07:00 (1 year, 11 months ago)
    
    CNAPI-725 test-task test failing because of a change introduced in CNAPI-724
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/lib/endpoints/tasks.js b/lib/endpoints/tasks.js
index 1aa5f58..6e8375b 100644
--- a/lib/endpoints/tasks.js
+++ b/lib/endpoints/tasks.js
@@ -83,15 +83,18 @@ Task.wait = function handlerTaskWait(req, res, next) {
         req.log.debug(
             'handlerTaskWait: task not complete nor failure, waiting on task');
 
-        wait(function (err) {
-            if (err) {
-                req.log.error(err, 'error waiting on task');
-
-                next(new restify.InternalError(
-                    'error waiting on task'));
-                return;
-            }
+        /**
+         * CNAPI conflates unsuccessful cn-agent task execution attempts with
+         * other types of non-task related errors (imagine receiving
+         * ECONNREFUSED whilst talking to cn-agent). In actuality, errors end
+         * up being inlined in the structure of the 'task' object.
+         *
+         * It would be useful to refactor CNAPI such that it enables us to
+         * differentiate between unsuccessful task requests and errors
+         * encountered during posting and waiting.
+         */
 
+        wait(function (err) {
             get(function () {
                 req.log.debug({ task: task },
                     'handlerTaskWait: got task after wait',
